From: Jason Dillaman <dillaman@redhat.com>
Date: Fri, 9 Sep 2016 08:31:52 -0400
Subject: librbd: potential seg fault when blacklisting an image client

Fixes: http://tracker.ceph.com/issues/17251
Signed-off-by: Jason Dillaman <dillaman@redhat.com>
(cherry picked from commit 7025fe8976f8672a2fce05ea693c06a8e9faed19)

Resolves: rhbz#1379837
---
 src/librbd/image_watcher/RewatchRequest.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/librbd/image_watcher/RewatchRequest.cc b/src/librbd/image_watcher/RewatchRequest.cc
index 4d808f4..cbf22c0 100644
--- a/src/librbd/image_watcher/RewatchRequest.cc
+++ b/src/librbd/image_watcher/RewatchRequest.cc
@@ -56,6 +56,7 @@ void RewatchRequest<I>::handle_unwatch(int r) {
   if (r == -EBLACKLISTED) {
     lderr(cct) << "client blacklisted" << dendl;
     finish(r);
+    return;
   } else if (r < 0) {
     lderr(cct) << "failed to unwatch: " << cpp_strerror(r) << dendl;
   }
